mlflow/utils/requirements_utils.py

```diff
@@ -417,7 +423,9 @@ def _infer_requirements(model_uri, flavor):
         *_MODULES_TO_PACKAGES.get("mlflow", []),
     ]
     packages = packages - set(excluded_packages)
-    unrecognized_packages = packages - _PYPI_PACKAGE_INDEX.package_names
+
+    # manually exclude mlflow[gateway] as it isn't listed separately in PYPI_PACKAGE_INDEX
+    unrecognized_packages = packages - _PYPI_PACKAGE_INDEX.package_names - {"mlflow[gateway]"}
```

Comment:
i felt a little weird about this, let me know if there's a better way to get the `gateway` extra recognized as a proper package.